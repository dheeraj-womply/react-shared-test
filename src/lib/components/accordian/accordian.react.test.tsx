import React from 'react'
import renderer from 'react-test-renderer'
import { AccordianSection } from './'
import { render, fireEvent } from '@testing-library/react';

test('Test snapshot for Accordian component', () => {
    const component = renderer.create(
        <AccordianSection 
            title={'Accordian Title'}>
                <div>div element</div>
        </AccordianSection>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
});


test('Test toggle for Accordian component', () => {
    const onClick = jest.fn()
    const { getByTestId } = render(
        <AccordianSection 
            toggle={onClick}
            title={'Accordian Title'}>
                <div>child element</div>
        </AccordianSection>
    );
    expect(getByTestId('accordian-title')).toHaveTextContent(/Accordian Title/i)
    expect(getByTestId('accordian-summary')).toHaveTextContent(/child element/i)
    fireEvent.click(getByTestId('accordian-title'));
    expect(onClick).toBeCalled();
});