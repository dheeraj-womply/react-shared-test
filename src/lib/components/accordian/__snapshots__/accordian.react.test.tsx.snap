// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot for Accordian component 1`] = `
<div
  className="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation0 MuiPaper-rounded"
>
  <div
    aria-disabled={false}
    aria-expanded={true}
    className="MuiButtonBase-root MuiAccordionSummary-root sc-AxirZ hZiiBu Mui-expanded"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content Mui-expanded"
    >
      <div
        className="sc-AxjAm fFIbso"
        data-testid="accordian-title"
      >
        Accordian Title
      </div>
    </div>
    <div
      aria-disabled={false}
      aria-hidden={true}
      className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role={null}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          role="region"
        >
          <div
            className="MuiAccordionDetails-root sc-AxiKw dJsaFy"
            data-testid="accordian-summary"
          >
            <div>
              div element
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
